parcels/include/parcels.h

```diff
@@ -208,9 +208,19 @@ static inline ErrorCode getCell2D(CField *f, int xi, int yi, int ti, float cell_
     float (*data_block)[zdim][ydim][xdim] = (float (*)[zdim][ydim][xdim]) f->data_chunks[blockid];
     for (tii=0; tii<2; ++tii){
       float (*data)[xdim] = (float (*)[xdim]) (data_block[ti+tii]);
-      for (yii=0; yii<2; ++yii)
-        for (xii=0; xii<2; ++xii)
-          cell_data[tii][yii][xii] = data[ilocal[0]+yii][ilocal[1]+xii];
+      for (yii=0; yii<2; ++yii){
+        for (xii=0; xii<2; ++xii){
+          if (xdim == 1)
+            xiid = 0;
+          else
+            xiid = xii;
+          if (ydim == 1)
+            yiid = 0;
+          else
+            yiid = yii;
+          cell_data[tii][yii][xii] = data[ilocal[0]+yiid][ilocal[1]+xiid];
```

Comment:
these if-statements for such a low-level, many-call function may inhibit performance as it cripples CPU branch prediction. A better formulation to achieve the same:
```
      int xiid = ((xdim==1) ? 1 : 0);
      int yiid = ((ydim==1) ? 1 : 0);
      for (yii=0; yii<2; ++yii)
        for (xii=0; xii<2; ++xii)
          cell_data[tii][yii][xii] = data[ilocal[0]+(yii*yiid)][ilocal[1]+(xii*xiid)];
```
from a machine instruction perspective, this is faster.

Reply:
Thanks, your suggestion indeed is much cleaner and nicer! My C is a bit rusty ;-)

Reply:
You're welcome. Also good that you spotted my mistake in the conditional statement